### PR TITLE
Iss26

### PR DESCRIPTION
**Pull request by cbravo135**
This still does not fully resolve issue 26. We still need to get the json configuration into all of the AnaProcessors.

### CODE REVIEW

**Comment by cbravo135** [comment]
Please double check that I didn't miss anything. We should discuss what we want to do with ParticleProcessor.

**Comment by cbravo135** [comment]
RefittedTracksProcessor and ClusterOnTrackAnaProcessor still need to be updated. RefittedTracksProcessor should be reorganized to use the new established scheme,
I think it is really an anaProcessor, and the nTuple with all the objects it needs could
have it's own new nTuple config file using the existing processors but using different
collection names.

**Comment by cbravo135** [comment]
ClusterOnTrackAnaProcessor is done. Just a few more words on what I am doing in this PR. We established a desired scheme with the last PR in how we would like the configuration of the processors to work in general. I have gone through and updated all of them to this new scheme. While I was at it, I also changed all the TClonesArrays to vectors, mostly because I didn't like what was used being dependent on the processor. I also updated EventProcessor to look for the different possibilities of trigger collections and parse them appropriately, so we now can handle 2016 and 2019 trigger data automatically.

**Comment by cbravo135** [comment]
Started looking at propagating the json histogram config to the rest of the HistoManagers and noticed that as the TrackHistos are now, we should consider separating out a VertexHistos from it. It seems right now all of our Vertex tools are folded into the Track tools, so we should soon make an effort to factorize this better. This won't be part of this PR just a note on where we should be looking in the near future.

**Comment by cbravo135** [comment]
MCAna and RecoHitAna are now updated to use JSON histogram configuration. This is probably where this PR should stop. Moving forward from here, we need to discuss how RefittedTracksProcessor should be factorized. The other feature we need in HistoManager is a way to easily generate one histogram for every hybrid in the tracker with only one entry in the json. We should be able to come up with a scheme to produce per hybrid histograms in a unified simplistic way.

File: processors/src/EventProcessor.cxx [hunk 1]
```diff
@@ -2,99 +2,138 @@
...
-    header.setSvtLatencyState(lc_event->getParameters().getIntVal("svt_latency_good")); 
+    header_->setSvtLatencyState(lc_event->getParameters().getIntVal("svt_latency_good")); 
 
     // Set the SVT position state
-    header.setSvtPositionState(lc_event->getParameters().getIntVal("svt_position_good"));
+    header_->setSvtPositionState(lc_event->getParameters().getIntVal("svt_position_good"));
 
     // Set the SVT event header state
-    header.setSvtEventHeaderState(lc_event->getParameters().getIntVal("svt_event_header_good"));
+    header_->setSvtEventHeaderState(lc_event->getParameters().getIntVal("svt_event_header_good"));
```
Comment (pbutti) [inline processors/src/EventProcessor.cxx:84]: Put a comment on this procedure, i.e. I try 2019 first and if data not availalbe it assumes 2016/5 is being processed.

**Review by pbutti — APPROVED** [review]

**Comment by pbutti** [comment]
Looks good to me. Can go in

**Pull request merged**